src/mimir_coordinator.py

```diff
@@ -259,6 +260,11 @@ def _build_blocks_storage_config(self) -> Dict[str, Any]:
         }
 
     def _build_s3_storage_config(self, s3_config_data: _S3ConfigData) -> Dict[str, Any]:
+        # Remove the scheme from the s3 endpoint for the Mimir config
+        parsed_url = urlparse(s3_config_data.endpoint)
+        scheme = "%s://" % parsed_url.scheme
+        s3_config_data.endpoint = parsed_url.geturl().replace(scheme, "", 1)
+        # Return the s3 configuration
```

Comment:
Does it make sense to move this to be a pydantic validator of the `endpoint` field in the `_S3ConfigData` class instead?

Reply:
The problem is I don't think we have control on what we receive over that relation

Reply:
sure but what I mean is 

```
    @validator("endpoint")
    def remove_scheme(cls, v):
        parsed_url = urlparse(v)
        scheme = "%s://" % parsed_url.scheme
        return parsed_url.geturl().replace(scheme, "", 1)
```

so regardless of what's coming in we remove the scheme, also we might need to accommodate for cases which the s3 endpoint will be an internal cluster storage (already has no scheme)

Reply:
Done!